aws-samples/amazon-transcribe-live-call-analytics
# Support PCA with Chime SDK Call Analytics

Update call_event_processor to process CallAnalyticsMetadata
KDS event generated by CallAnalytics. The call_event_processor
sends an EventBridge event to chime_call_analytics_initialization.
The EventBridge event contains the mapping of the call id to session id.
chime_call_analytics_initialization creates the SessionData
for pca_integration and process_tca_postcall Lambdas.
pca_integration and process_tca_postcall Lambdas are updated to
get the transcript and media path from the TCA completion event.

Test
manually tested with a LCA call

Migrate to StartVoiceToneAnalysisTask

Migrate to the new StartVoiceToneAnalysisTask API in the media pipeline
namespace. Using the new API allows LCA to have a single
MediaInsightPipelineConfiguration rather the workaround with
two MediaInsightPipelineConfigurations.

Test
Manually update CF template to a single MediaInsightPipelineConfiguration.
Manually tested voice tone analysis and review the logs in Voice Tone
Analysis lambda. Validate that voice tone fluctuation is shown
in LCA console.

Support existing VC Id instead of creating VC

Allow customer to specify their own Voice Connector as a CloudFormation
Template parameter. If the customer provided Voice Connector is provided,
LCA will not deploy a new VC. EventBridgeRule for LCA Lambda will
use the provided VC Id.

Test
deploy CFT and run VC

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
